src/main/java/seedu/address/model/order/Item.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.model.order;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the item description of a Person's Order in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidItem(String)}
+ */
+public class Item {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Item should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String fullDescription;
+
+    /**
+     * Constructs a {@code Item}.
+     *
+     * @param item A valid item description.
+     */
+    public Item(String item) {
+        requireNonNull(item);
+        checkArgument(isValidItem(item), MESSAGE_CONSTRAINTS);
+        fullDescription = item;
+    }
+
+    /**
+     * Returns true if a given string is a valid item description.
+     */
+    public static boolean isValidItem(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+
```

Comment:
LGTM. Small nitpick: can we remove this extra line here?

Reply:
Noted. Change is completed in update.